### PR TITLE
Bulk vendor add table ux fixes

### PR DESCRIPTION
**Pull request by TheAndrewJackson**
Closes [#PROD-1292](https://ethyca.atlassian.net/browse/PROD-1292)

### Description Of Changes

Improve the UX of the bulk vendor add table

Screenshot 2023-11-14 at 15 25 20

UAT ENV here: https://fides-plus-nightly-fmg5dfemx-ethyca.vercel.app/add-systems/multiple

### Code Changes

* [ ] Fix table header height on safari
* [ ] add commas to all number strings
* [ ] Move the add vendors better to the right of the search bar
* [ ] update the row selection bar to only show up when clicking the header checkbox
* [ ] sort the rows by default
* [ ] Add number to filter button to display the total amount of filters applied

### Steps to Confirm

* [ ] Go to the bulk vendor add page
* [ ] Walk through the AC in the Jira ticket and make sure all of it has been done

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/admin-ui/src/features/common/table/v2/FidesTable.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ export const FidesTableV2 = <T,>({
         backgroundColor="gray.50"
       >
         {tableInstance.getHeaderGroups().map((headerGroup) => (
-          <Tr key={headerGroup.id}>
+          <Tr key={headerGroup.id} height="inherit">
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/common/table/v2/FidesTable.tsx:74]: this fixes the header height on safari

File: clients/admin-ui/src/features/common/table/v2/PaginationBar.tsx [hunk 2]
```diff
@@ -47,8 +47,11 @@ export const PaginationBar = <T,>({
               fontVariantNumeric: "tabular-nums",
             }}
           >
-            {startRange}-{endRange <= totalRows ? endRange : totalRows} of{" "}
-            {totalRows}
+            {startRange.toLocaleString("en")}-
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/common/table/v2/PaginationBar.tsx:50]: this adds commas to numbers that are larger than 999

File: clients/admin-ui/src/features/plus/plus.slice.ts [hunk 3]
```diff
@@ -563,5 +563,26 @@ export type DictSystems = {
...
   [(RootState) => RootState, plusApi.endpoints.getAllSystemVendors.select()],
-  (RootState, { data }) => data || EMPTY_DICT_SYSTEMS
+  (RootState, { data }) =>
+    data
+      ? data
+          .slice()
+          .map((ds) => {
+            const name = ds.name
+              .split(" ")
+              .map((word) =>
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/plus/plus.slice.ts:573]: this makes it so legal names from compass that are in all caps aren't in all caps. This fixes an UX issue where all of the all caps names go to the top when the vendors are sorted

File: clients/admin-ui/src/features/system/add-multiple-systems/AddMultipleSystems.tsx [hunk 4]
```diff
@@ -102,13 +107,22 @@ export const AddMultipleSystems = ({ redirectRoute }: Props) => {
         header: ({ table }) => (
           <IndeterminateCheckboxCell
             {...{
-              checked: table.getIsAllRowsSelected(),
-              indeterminate:
+              dataTestId: "select-page-checkbox",
+              checked: table.getIsAllPageRowsSelected(),
+              indeterminate: table
+                .getPaginationRowModel()
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/add-multiple-systems/AddMultipleSystems.tsx:113]: this updates the header checkbox to only select the currently displayed page.

File: clients/admin-ui/src/features/system/add-multiple-systems/AddMultipleSystems.tsx [hunk 5]
```diff
@@ -246,6 +260,14 @@ export const AddMultipleSystems = ({ redirectRoute }: Props) => {
     .getSelectedRowModel()
     .rows.filter((r) => !r.original.linked_system).length;
 
+  const totalFilters =
+    tableInstance.getState().columnFilters.length > 0
+      ? // @ts-ignore
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/system/add-multiple-systems/AddMultipleSystems.tsx:265]: For some reason TS thinks the `tableInstance` below is `unknown`. I wasn't able to make it happy

File: clients/admin-ui/src/features/plus/plus.slice.ts (hunk 3 above)
Comment (jpople) [reply clients/admin-ui/src/features/plus/plus.slice.ts:573]: Small nitpick: I feel like I'd prefer for this to be in like a `toTitleCase` helper method somewhere, it's pretty hard to parse in this form and location.

**Review by jpople — APPROVED** [review]
Looking solid to me!
